### PR TITLE
Modify Add job command

### LINKED ISSUE #71: Make certain parameters optional for the add_job command
Currently, the iteration of add_job is only able to add_jobs with ALL of the parameters. 

As all the features have not been stabilized yet (feature in mind "screen"), plan to handle this in a later iteration.

### PR DESCRIPTION
**Pull request by Quasant**
## Changes
- Attribute `JobRequirement requirements` changed to `Set<Tags> requirements`
- Now only `Requirements` is optional, the others are mandatory
- Bug fixed, closes #71 

## Issues faced, observations and potential improvements
- The `FlowPane` or the area of the tags in the UI lost the styling 
- Contacts has a `set<Skills>` field. Jobs has a `set<Tags>` field. Propose to standardize the class used since `Skills` is essentially `Tags`
  - This might make the life of the implementer of the `screen` command easier
    - We can match `Tags` with other `Tags` 
    - If need be, I can make `Description` in `Job` equivalent to `Role` in `Person`, or leave it as a string
      -  Implementer simply has to do something akin to `Description.contains(Role)` 

## Potential tasks
- Remove certain classes that are now unused (e.g. JobRequirements)

### CODE REVIEW

File: src/main/java/seedu/address/model/job/Job.java [hunk 1]
```diff
@@ -9,20 +15,21 @@ public class Job {
     private final JobName name;
     private final JobCompany company;
     private final JobSalary salary;
-    private final JobRequirements requirements;
     private final JobDescription description;
+    private final Set<Tag> requirements = new HashSet<>();
 
     /**
      * TODO: Implement data validation
+     * TODO: What is meant by this ah ^^?
```
Comment (Quasant) [inline src/main/java/seedu/address/model/job/Job.java:23]: Just in case this is missed

**Review by KengHian — APPROVED** [review]
Some nitpicks, otherwise LGTM.

File: src/main/java/seedu/address/model/job/Job.java [hunk 2]
```diff
@@ -37,14 +44,18 @@ public JobSalary getSalary() {
...
-    }
-
     public JobDescription getDescription() {
         return description;
     }
 
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
```
[the remark this answers is not in the document]
Comment (Quasant) [reply src/main/java/seedu/address/model/job/Job.java:54]: It returns the requirements for the job only, if I am understanding your 2nd part of the comment correctly. 

If you want to extract it out as a method, can be done, but we should standardise to using a specific class.

File: src/main/java/seedu/address/model/job/Job.java (hunk 1 above)
Comment (Quasant) [reply src/main/java/seedu/address/model/job/Job.java:23]: Ah I see. 

It's only for `JobCompany` since we haven't integrated.

**Review by Green-Tea-123 — APPROVED** [review]
LGTM

**Pull request merged**